src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,573 @@
+        return CHIP_NO_ERROR;
+    }
+
+    void Clear() override
+    {
+        scene_count = 0;
+        for (uint8_t i = 0; i < kMaxScenesPerFabric; i++)
+        {
+            scene_map[i].Clear();
+        }
+    }
+
+    CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override
+    {
+        TLV::TLVType container;
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, container));
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(TagScene::kSceneCount), scene_count));
+        ReturnErrorOnFailure(writer.StartContainer(TLV::ContextTag(TagScene::kStorageIDArray), TLV::kTLVType_Array, container));
+
+        // Storing the scene map
+        for (uint8_t i = 0; i < kMaxScenesPerFabric; i++)
+        {
+            ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, container));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(TagScene::kEndpointID), (scene_map[i].mEndpointId)));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(TagScene::kGroupID), (scene_map[i].mGroupId)));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(TagScene::kSceneID), (scene_map[i].mSceneId)));
+            ReturnErrorOnFailure(writer.EndContainer(container));
+        }
+        ReturnErrorOnFailure(writer.EndContainer(container));
+        return writer.EndContainer(container);
+    }
+
+    CHIP_ERROR Deserialize(TLV::TLVReader & reader) override
+    {
+        ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+
+        TLV::TLVType container;
+        ReturnErrorOnFailure(reader.EnterContainer(container));
+
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kSceneCount)));
+        ReturnErrorOnFailure(reader.Get(scene_count));
+        ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::ContextTag(TagScene::kStorageIDArray)));
+        ReturnErrorOnFailure(reader.EnterContainer(container));
+
+        uint8_t i = 0;
+        CHIP_ERROR err;
+        while ((err = reader.Next(TLV::AnonymousTag())) == CHIP_NO_ERROR && i < kMaxScenesPerFabric)
+        {
+            ReturnErrorOnFailure(reader.EnterContainer(container));
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kEndpointID)));
+            ReturnErrorOnFailure(reader.Get(scene_map[i].mEndpointId));
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kGroupID)));
+            ReturnErrorOnFailure(reader.Get(scene_map[i].mGroupId));
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kSceneID)));
+            ReturnErrorOnFailure(reader.Get(scene_map[i].mSceneId));
+            ReturnErrorOnFailure(reader.ExitContainer(container));
+
+            i++;
+        }
+        VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
+
+        ReturnErrorOnFailure(reader.ExitContainer(container));
+        return reader.ExitContainer(container);
```

Comment:
Again, the `container` variables should be separate, one per container we are dealing with.